### PR TITLE
MM-18900 - Remove unused status flags in Search request state object

### PR DESCRIPTION
**Pull request by nadalfederer**
#### Summary
Remove unused status flags in Search request state object
#### Ticket Link

 one part to close https://github.com/mattermost/mattermost-server/issues/12522

### CODE REVIEW

**Comment by nadalfederer** [comment]
there are some views in matter-mobile which depend on these flags. It will take some time to refactor them.

**Comment by sudheerDev** [comment]
Sorry about the delay @nadalfederer. Would you be open to merging master and resolving the conflicts?

**Comment by nadalfederer** [comment]
Hi, sure, I'll resolve conflict

File: src/reducers/entities/search.ts [hunk 1]
```diff
@@ -262,30 +262,6 @@ function current(state: any = {}, action: GenericAction) {
...
-    switch (action.type) {
-    case SearchTypes.SEARCH_POSTS_REQUEST:
-        return action.isGettingMore;
-    case SearchTypes.SEARCH_POSTS_FAILURE:
-    case SearchTypes.SEARCH_POSTS_SUCCESS:
-        return false;
-    default:
-        return state;
-    }
-}
```
Comment (saturninoabril) [inline src/reducers/entities/search.ts]: These functions are being used by webapp and mobile.  Please take a look again.

**Comment by jasonblais** [comment]
Hey @nadalfederer! Let us know if you have any questions about the feedback above?

If you're busy, no worries, just let us know. Really appreciate your contributions past months!!

**Comment by nadalfederer** [comment]
Hi, I'm sorry for the delay. I'll fix current issues soon

**Comment by jasonblais** [comment]
No worries, thanks @nadalfederer 🎉

**Comment by nadalfederer** [comment]
As @saturninoabril mentioned above. isSearchingTerm, isSearchGettingMore are being used in webapp and mobile. So, I think, they cannot be removed because there are params that are being used in redux, and move them to webapp don't seem to be likely

**Comment by hanzei** [comment]
@nadalfederer Heads up that there is a merge conflict with your PR

**Comment by sudheerDev** [comment]
@nadalfederer It is ok to leave them as they are. It is fine. It is not a pattern we can completely get away with.

**Pull request merged**